### PR TITLE
feat: contacts

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

**Review by michaelsbradleyjr — COMMENTED** [review]
In the process of learning about how to create the CRUD APIs by looking at e.g. this PR, I realized the table creation code is built up from `CREATE TABLE` and `ALTER TABLE` sql statements in status-go.

I see that in `status-go/protocol/migrations/sqlite/000002_add_last_ens_clock_value.up.sql` there is a statement:
```sql
ALTER TABLE contacts ADD COLUMN last_ens_clock_value INT NOT NULL DEFAULT 0;
```

But the `last_ens_clock_value` column in status-go has no counterpart in this PR and I wasn't sure why that's the case; was it accidentally omitted?

File: nim_status/lib/contacts.nim [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    identicon* {.serializedFieldName($ContactType.Identicon), dbColumnName($ContactsCol.Identicon).}: string 
+    # base64 encoded photo
+    photo* {.serializedFieldName($ContactType.Photo), dbColumnName($ContactsCol.Photo).}: Option[string]
+    # last time we received an update from the contact. Updates should be discarded if last updated is less than the one stored
+    lastUpdated* {.serializedFieldName($ContactType.LastUpdated), dbColumnName($ContactsCol.LastUpdated).}: int64
+    # contains information about whether we blocked/added/have been added
+    systemTags* {.serializedFieldName($ContactType.SystemTags), dbColumnName($ContactsCol.SystemTags).}: seq[string]
+    deviceInfo* {.serializedFieldName($ContactType.DeviceInfo), dbColumnName($ContactsCol.DeviceInfo).}: seq[ContactDeviceInfo]
+    tributeToTalk* {.serializedFieldName($ContactType.TributeToTalk), dbColumnName($ContactsCol.TributeToTalk).}: Option[string]
+    localNickname* {.serializedFieldName($ContactType.TributeToTalk), dbColumnName($ContactsCol.TributeToTalk).}: Option[string]
```
Comment (michaelsbradleyjr) [inline nim_status/lib/contacts.nim:87]: Is e.g. `ContactType.TributeToTalk` correct for `localNickname`?

Comment (richard-ramos) [reply nim_status/lib/contacts.nim:87]: No. `localNickname` is a `string`

**Comment by richard-ramos** [comment]
> In the process of learning about how to create the CRUD APIs by looking at e.g. this PR, I realized the table creation code is built up from `CREATE TABLE` and `ALTER TABLE` sql statements in status-go.
> 
> I see that in `status-go/protocol/migrations/sqlite/000002_add_last_ens_clock_value.up.sql` there is a statement:
> 
> ```sql
> ALTER TABLE contacts ADD COLUMN last_ens_clock_value INT NOT NULL DEFAULT 0;
> ```
> 
> But the `last_ens_clock_value` column in status-go has no counterpart in this PR and I wasn't sure why that's the case; was it accidentally omitted?

Interesting!  It seems that table column is not used at all? https://github.com/status-im/status-go/blob/develop/protocol/persistence.go I could not find references to it

**Comment by emizzle** [comment]
> > In the process of learning about how to create the CRUD APIs by looking at e.g. this PR, I realized the table creation code is built up from `CREATE TABLE` and `ALTER TABLE` sql statements in status-go.
> > I see that in `status-go/protocol/migrations/sqlite/000002_add_last_ens_clock_value.up.sql` there is a statement:
> > ```sql
> > ALTER TABLE contacts ADD COLUMN last_ens_clock_value INT NOT NULL DEFAULT 0;
> > ```
> > 
> > 
> > But the `last_ens_clock_value` column in status-go has no counterpart in this PR and I wasn't sure why that's the case; was it accidentally omitted?
> 
> Interesting! It seems that table column is not used at all? https://github.com/status-im/status-go/blob/develop/protocol/persistence.go I could not find references to it

I ended up adding this column back in just so that we're on parity with status-go.

**Pull request merged**